packages/liquidation-sdk-viem/src/preLiquidation/positionGetters.ts

```diff
@@ -4,69 +4,99 @@ import { fetchAccrualPosition } from "@morpho-org/blue-sdk-viem";
 import { Time } from "@morpho-org/morpho-ts";
 import type { Account, Chain, Client, Transport } from "viem";
 import { apiSdk } from "../api";
-import { authorizationLogs, preLiquidationLogs } from "./logGetters";
-import { type PreLiquidation, PreLiquidationPosition } from "./types";
+import {
+  type PreLiquidation,
+  type PreLiquidationData,
+  PreLiquidationPosition,
+} from "./types";
 
 export async function getPreLiquidablePositions(
   client: Client<Transport, Chain, Account>,
   whitelistedMarkets: MarketId[],
 ) {
   const chainId = client.chain.id;
 
-  const preLiquidations = (await preLiquidationLogs(client)).filter(
-    (preLiquidation) => whitelistedMarkets.includes(preLiquidation.marketId),
-  );
+  const url = `${process.env.INDEXER_API_URL}/chain/${chainId}/preliquidations`;
 
-  const preLiquidationInstances = await Promise.all(
-    preLiquidations.map(async (preLiquidation) => {
-      const {
-        markets: { items: market },
-      } = await apiSdk.getMarketAssets({
-        chainId,
-        marketId: preLiquidation.marketId,
-      });
+  try {
+    const response = await fetch(url, {
+      method: "POST",
+      body: JSON.stringify({ marketIds: whitelistedMarkets }),
+    });
 
-      const loanAsset = market !== null ? market[0]?.loanAsset : undefined;
-      const collateralAsset =
-        market !== null ? market[0]?.collateralAsset : undefined;
+    const data = (await response.json()) as {
+      results: PreLiquidationData[];
+    };
 
-      if (
-        loanAsset === undefined ||
-        collateralAsset === undefined ||
-        collateralAsset === null
-      )
-        return;
+    const preLiquidationInstances = await Promise.all(
+      data.results
+        .filter((preLiquidation) => preLiquidation.price !== null)
+        .map((preLiquidation) => {
+          return {
+            ...preLiquidation,
+            price: BigInt(preLiquidation.price!),
+            preLiquidationParams: {
+              ...preLiquidation.preLiquidationParams,
+              preLCF1: BigInt(preLiquidation.preLiquidationParams.preLCF1),
+              preLCF2: BigInt(preLiquidation.preLiquidationParams.preLCF2),
+              preLIF1: BigInt(preLiquidation.preLiquidationParams.preLIF1),
+              preLIF2: BigInt(preLiquidation.preLiquidationParams.preLIF2),
+              preLltv: BigInt(preLiquidation.preLiquidationParams.preLltv),
+            },
+          };
+        })
+        .map(async (preLiquidation) => {
+          const {
+            markets: { items: market },
+          } = await apiSdk.getMarketAssets({
+            chainId,
+            marketId: preLiquidation.marketId,
+          });
```

Comment:
We're making N API calls here, where N is the nb of markets
Instead, group them all in a single API query to decrease the API and bandwidth consumption